### PR TITLE
Migrate tests

### PR DESCRIPTION
**Pull request by aashu16**
The primary goals of this PR are:
- Replace Karma/Jasmine with Vitest
- Migrate tests from Enzyme to @testing-library/react

Checklist:
- [x] Configure vitest
- [x] Update test utils
- [x] Update test files
   - [x] input_numeric_format
   - [x] keypress_and_caret
   - [x] utils
   - [x] input
   - [x] format_as_text
- [x] Remove packages that are no longer required - Karma, Jasmine, Enzyme + related dependencies

### CODE REVIEW

File: .yarn/releases/yarn-1.22.19.cjs
Comment (s-yadav) [inline .yarn/releases/yarn-1.22.19.cjs:1]: Looks like this got staged and after that .yarn folder is added on gitignore. You will need to untrack the file.

File: package.json [hunk 1]
```diff
@@ -48,25 +49,23 @@
...
+    "@docusaurus/core": "2.0.0-beta.18",
+    "@docusaurus/preset-classic": "2.0.0-beta.18",
+    "@mdx-js/react": "^1.6.22",
     "@rollup/plugin-buble": "^0.21.3",
     "@rollup/plugin-typescript": "^8.3.0",
-    "@testing-library/react": "^12.1.2",
-    "@testing-library/react-hooks": "8.0.1",
-    "@testing-library/user-event": "^13.5.0",
+    "@testing-library/react": "^13.0.0",
+    "@testing-library/user-event": "^14.0.0",
```
Comment (s-yadav) [inline package.json:59]: With latest testing-library few test were breaking which works on browser. not sure if you faced same issue.

File: .yarn/releases/yarn-1.22.19.cjs
Comment (aashu16) [reply .yarn/releases/yarn-1.22.19.cjs:1]: Yarn docs [recommend](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored) committing `.yarn/releases` to the repo. The reasoning behind this is:

> You will want to keep them _(.yarn/plugins and .yarn/releases)_ versioned (this prevents potential issues if, say, two engineers use different Yarn versions with different features).

Setting yarn version to 3.x should remove this but doing it now will also make significant changes to the lockfile. This is here for the moment to use a Yarn classic lock file with Yarn 3.x. I will try and sort this out before converting this into a PR.

File: package.json (hunk 1 above)
Comment (aashu16) [reply package.json:59]: Haven't noticed any issues like that so far, but I might have missed something.

There are a handful of tests that fail in `input_numeric_format.spec.js` using the updated test utils. The last time I checked (about a month ago), I could reproduce the failed tests in Chrome 113 and Safari 16.x.

Comment (s-yadav) [reply package.json:59]: Cool lets try updating, I tried updating it last time I was fixing caret position issue, and got some issue so did not do upgrade.

Comment (aashu16) [reply package.json:59]: Sounds good. Will try and finish updating the remaining tests by the 18th.

File: .yarn/releases/yarn-1.22.19.cjs
Comment (aashu16) [reply .yarn/releases/yarn-1.22.19.cjs:1]: The `.yarn/releases` directory contains whatever version of Yarn is being used. I can ignore it, but it is not recommended.

**Comment by s-yadav** [comment]
Merging this to the dev will work with main repo directly.

**Pull request merged**